File: Java/BinaryTree/BinaryTreeTraversal.java
/*
*
*   @author
*   Aakash Verma
*
*   Output:
*   Pre Order Traversal is: 1 2 4 5 3 6 7 
*   In Order Traversal is: 4 2 5 1 6 3 7 
*   Post Order Traversal is: 4 5 2 6 7 3 1 
*
*/


/*  
    Creating a structure for the node.
    Initializing the node's data upon calling its constructor.
*/

class Node {
    int data;
    Node left, right;
    
    public Node(int key) {
        data = key;
        left = right = null;
    }
}


/* Defining class for the Binary Tree */
class BinaryTreeTraversal {

    /* Creating a root node for the Tree */
    static Node root;
    

    /* 
        Assigning root as null initially. So as soon as the object will be created 
        of this BinaryTreeTraversal class, root will be set as null.
     */
    BinaryTreeTraversal() {
        root = null;
    }
    
    /* Pre Order Traversal */
    void preOrder(Node root) {
        if(root == null) {
            return;
        }
        System.out.print(root.data + " ");
        preOrder(root.left);
        preOrder(root.right);
    }


    /* In Order Traversal */
    void inOrder(Node root) {
        if(root == null) {
            return;
        }
        inOrder(root.left);
        System.out.print(root.data + " ");
        inOrder(root.right);
    }

    /* Post Order Traversal */
    void postOrder(Node root) {
        if(root == null) {
            return;
        }
        postOrder(root.left);
        postOrder(root.right);
        System.out.print(root.data + " ");
    }
    
    /* main method */
    public static void main (String[] args) {
        BinaryTreeTraversal tree = new BinaryTreeTraversal();
        tree.root = new Node(1);
        tree.root.left = new Node(2);
        tree.root.right = new Node(3);
        tree.root.left.left = new Node(4);
        tree.root.left.right = new Node(5);
        tree.root.right.left = new Node(6);
        tree.root.right.right = new Node(7);
        System.out.print("Pre Order Traversal is: ");
        tree.preOrder(root);
        System.out.println();
        System.out.print("In Order Traversal is: ");
        tree.inOrder(root);
        System.out.println();
        System.out.print("Post Order Traversal is: ");
        tree.postOrder(root);
        System.out.println();
    }
}

File: Python/BinaryTree/BinaryTreeTraversal.py

# @author
# Aakash Verma

# Output:
# Pre Order Traversal is: 1 2 4 5 3 6 7 
# In Order Traversal is: 4 2 5 1 6 3 7 
# Post Order Traversal is: 4 5 2 6 7 3 1 


# Creating a structure for the node.
# Initializing the node's data upon calling its constructor.

class Node:
    def __init__(self, data):
        self.data = data
        self.right = self.left = None


# Defining class for the Binary Tree
class BinaryTreeTraversal:
    
    # Assigning root as null initially. So as soon as the object will be created 
    # of this BinaryTreeTraversal class, root will be set as null.
    def __init__(self): 
        self.root = None
    
    # Pre Order Traversal
    def preOrder(self, root):
        if(root is None):
            return
        print(root.data, end = " ")
        self.preOrder(root.left)
        self.preOrder(root.right)


    # In Order Traversal
    def inOrder(self, root):
        if(root is None):
            return
        self.inOrder(root.left)
        print(root.data, end = " ")
        self.inOrder(root.right)

    # Post Order Traversal */
    def postOrder(self, root):
        if(root is None):
            return
        self.postOrder(root.left)
        self.postOrder(root.right)
        print(root.data, end = " ")
    
#main method
if __name__ == '__main__':
    tree = BinaryTreeTraversal()
    tree.root = Node(1)
    tree.root.left = Node(2)
    tree.root.right = Node(3)
    tree.root.left.left = Node(4)
    tree.root.left.right = Node(5)
    tree.root.right.left = Node(6)
    tree.root.right.right = Node(7)
    print("Pre Order Traversal is:",  end = " ")
    tree.preOrder(tree.root)
    print()
    print("In Order Traversal is:", end = " ")
    tree.inOrder(tree.root)
    print()
    print("Post Order Traversal is:", end = " ")
    tree.postOrder(tree.root)
    print()
